docs/Errors.md

```diff
@@ -4,18 +4,49 @@
 ## Errors
 
 <a name="error-handling"></a>
-### Error Handling
+### Error Handling In Node.js
 
-Uncaught errors are likely to cause memory leaks, file descriptor leaks and other major production issues. [Domains](https://nodejs.org/en/docs/guides/domain-postmortem/) were introduced to try fixing this issue, but they did not. Given the fact that it is not possible to process all uncaught errors sensibly, the best way to deal with them at the moment is to [crash](https://nodejs.org/api/process.html#process_warning_using_uncaughtexception_correctly). In case of promises, make sure to [handle](https://nodejs.org/dist/latest-v8.x/docs/api/deprecations.html#deprecations_dep0018_unhandled_promise_rejections) errors [correctly](https://github.com/mcollina/make-promises-safe).
+#### Uncaught Errors
+In node.js, uncaught errors are likely to cause memory leaks, file descriptor leaks and other major production issues.  [Domains](https://nodejs.org/en/docs/guides/domain-postmortem/) were introduced to try fix this issue, but did not. 
 
-Fastify follows an all-or-nothing approach and aims to be lean and optimal as much as possible. Thus, the developer is responsible for making sure that the errors are handled properly. Most of the errors are usually a result of unexpected input data, so we recommend specifying a [JSON.schema validation](./Validation-and-Serialization.md) for your input data.
+Given that it is not possible to process all uncaught errors sensibly, the best way to deal with them is to [crash](https://nodejs.org/api/process.html#process_warning_using_uncaughtexception_correctly). 
 
-Fastify tries to catch as many uncaught errors it can without hindering performance. This includes:
+#### Catching Errors In Promises
+In case of promises, make sure to [handle](https://nodejs.org/dist/latest-v8.x/docs/api/deprecations.html#deprecations_dep0018_unhandled_promise_rejections) errors correctly [by making promises safe](https://github.com/mcollina/make-promises-safe).
```

Comment:
I would not link to the `unhandledRejection` deprecation notice. 

You might want to lift some of the motivations from make-promises-safe instead.